src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -4086,8 +4094,8 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         case Apply(Select(New(_), nme.CONSTRUCTOR), _) if pt.typeSymbol == ArrayClass =>
           reportAnnotationError(ArrayConstantsError(tree)); None
 
-        case ann @ Apply(Select(New(tpt), nme.CONSTRUCTOR), _) if isJava =>
-          val annInfo = typedAnnotation(ann, None, mode)
+        case Apply(Select(New(tpt), nme.CONSTRUCTOR), _) if isJava =>
+          val annInfo = typedAnnotation(tree, None, mode)
```

Comment:
this edit misses a second usage a few lines later. The real danger is not shadowing but unshadowing.

Reply:
IDE could help with "atomic" refactor, but also incremental compilation could notice a change in attribution. (Previously, I've wanted incremental compilation to track something; I haven't kept a list.)